vortex-alp/src/compress.rs

```diff
@@ -34,11 +29,11 @@ macro_rules! match_each_alp_float_ptype {
 impl EncodingCompression for ALPEncoding {
     fn can_compress(
         &self,
-        array: &dyn Array,
+        array: &Array,
         _config: &CompressConfig,
     ) -> Option<&dyn EncodingCompression> {
         // Only support primitive arrays
-        let parray = array.maybe_primitive()?;
+        let parray = array.clone().flatten_primitive().unwrap();
```

Comment:
Before, this was just an attempted downcast. Now you're cloning and decompressing.

The new equivalent is `PrimitiveArray::try_from(array)`